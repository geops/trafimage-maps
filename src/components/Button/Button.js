/* eslint-disable jsx-a11y/no-access-key */
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * Function triggered on button's click event.
   */
  onClick: PropTypes.func.isRequired,

  /**
   * Children content of the button.
   */
  children: PropTypes.node,

  /**
   * CSS class of the button.
   */
  className: PropTypes.string,

  /**
   * HTML tabIndex attribute
   */
  tabIndex: PropTypes.number,

  /**
   * Title of the button.
   */
  title: PropTypes.string,

  /**
   * Keyboard accesskey shortcut.
   */
  accessKey: PropTypes.string,

  /**
   * Arai global attribute that indicates if the element is expandable.
   */
  ariaExpanded: PropTypes.bool,

  /**
   * HTML style attribute
   */
  style: PropTypes.object,

  /**
   * HTML disabled attribute
   */
  disabled: PropTypes.bool,
};

const defaultProps = {
  className: 'tm-button',
  children: undefined,
  title: undefined,
  tabIndex: 0,
  style: undefined,
  accessKey: undefined,
  ariaExpanded: undefined,
  disabled: undefined,
};

/**
 * This component displays a simple button.
 */
class Button extends PureComponent {
  render() {
    const {
      onClick,
      children,
      className,
      title,
      tabIndex,
      style,
      accessKey,
      ariaExpanded,
      disabled,
    } = this.props;

    return (
      <div
        role="button"
        style={style}
        tabIndex={tabIndex}
        className={className}
        title={title}
        aria-label={title}
        aria-expanded={ariaExpanded}
        onClick={(e) => {
          if (disabled) {
            return;
          }
          onClick(e);
        }}
        onKeyPress={(e) => e.which === 13 && onClick(e)}
        accessKey={accessKey}
        disabled={disabled}
      >
        {children}
      </div>
    );
  }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
