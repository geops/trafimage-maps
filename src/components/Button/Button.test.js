import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Button from "./Button";

const funcs = {
  onClick: () => {},
};

test("Button should match snapshot.", () => {
  const { container } = render(
    <Button
      className="tm-zoom in"
      title="Zoom in"
      onClick={() => funcs.onClick()}
    >
      +
    </Button>,
  );
  expect(container.innerHTML).toMatchSnapshot();
});

test("Button should update.", () => {
  const spy = jest.spyOn(funcs, "onClick");
  const { container } = render(
    <Button className="tm-class" title="Zoom" onClick={() => funcs.onClick()}>
      +
    </Button>,
  );

  fireEvent.click(container.querySelector(".tm-class"));
  fireEvent.keyPress(container.querySelector(".tm-class"), {
    key: "Enter",
    code: "Enter",
    charCode: 13,
    which: 13,
  });

  expect(spy).toHaveBeenCalledTimes(2);
});
