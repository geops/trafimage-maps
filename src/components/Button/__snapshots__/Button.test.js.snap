// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot. 1`] = `
<div role="button"
     tabindex="0"
     class="tm-zoom in"
     title="Zoom in"
     aria-label="Zoom in"
>
  +
</div>
`;
